classes/StaffManager.php

```diff
@@ -513,60 +517,77 @@ public function canIssueOW($staff_level) {
     }
 
     public function banUser($ban_type, $ban_length, $user_data) {
-        if($ban_length > 0 || $ban_length == self::PERM_BAN_VALUE) {
-            $ban_data = json_decode($user_data['ban_data'], true);
-            $ban_expire = ($ban_length == self::PERM_BAN_VALUE ? $ban_length : time() + ($ban_length * 86400));
-            $old_ban_remaining = false;
-
-            //Create new ban data for storage
-            if($ban_data == null || empty($ban_data)) {
-                $ban_data = json_encode(array(
-                    $ban_type => $ban_expire
-                ));
+        $ban_data = json_decode($user_data['ban_data'], true);
```

Comment:
Follow-up request, can we get a DTO for ban data with a static method to initialize from db data?

Reply:
I was thinking about doing this with the post-character staff panel upgrades that are needed. GMTA